src/main/java/hoodini/Parser.java

```diff
@@ -0,0 +1,278 @@
+                    str1.split("from:")[1].split(" to: ")[0],
+                    str1.split("to:")[1].replace(")", ""));
+            store.add(event);
+
+        }
+    }
+
+    /**
+     * Method to read from file and input content
+     * to the chatbot
+     * @param filepath the path of the file
+     * @throws ReadException Handles invalid tasks in file
+     */
+
+    public void readFromFile(String filepath) throws ReadException {
+        try {
+            java.io.File file = new java.io.File(filepath);
+            java.util.Scanner input = new java.util.Scanner(file);
+            while (input.hasNext()) {
+                String str = input.nextLine();
+                if (str.startsWith("[T]")) {
+                    handleFileToDo(str.substring(4));
+                } else if (str.startsWith("[D]")) {
+                    handleFileDeadline(str.substring(4));
+                } else if (str.startsWith("[E]")) {
+                    handleFileEvent(str.substring(4));
+                } else {
+                    throw new ReadException("Whoopsie! "
+                            + "There are invalid tasks in the file");
+                }
+            }
+        } catch (java.io.FileNotFoundException e) {
+            System.out.println("Whoopsie! File not found");
+
+        }
+    }
+
+
+    /**
+     * Method to end the chatbot
+     */
+    private String end() {
+        try {
+            String home = System.getProperty("user.home");
+            String filePath = home + "/Desktop/hoodini.txt";
+            store.writeToFile(filePath);
+            return ui.showGoodbye();
+        } catch (IOException e) {
+            return "Whoopsie! There was an error writing to the file";
+        }
+    }
+
+    /**
+     * Handles the user inputs and identifies tasks to store
+     * into the task list
+     */
+    public String handleInput(String input) {
+        assert input != null : "Input should not be null";
+        try {
+            String command = input.split(" ")[0];
+            Case cases = Case.getCase(command);
+            switch (cases) {
+            case BYE:
+                return end();
+            case LIST:
+                return store.output();
+            case TODO:
+                return handleToDo(input);
+            case DEADLINE:
+                return handleDeadline(input);
+            case EVENT:
+                return handleEvent(input);
+            case MARK:
+                return handleMark(input);
+            case UNMARK:
+                return handleUnmark(input);
+            case DELETE:
+                return handleDelete(input);
+            case FIND:
+                return handleFind(input);
+            case SORT:
+                return store.sort();
+            case ERROR:
+            default:
+                throw new InvalidInputException("Whoopsie! "
+                        + "I am unable to understand your request!");
```

Comment:
I like that you separated the logic of each case into separate functions. Makes the code here look very clean